app/assets/javascripts/actions/article_actions.js

```diff
@@ -3,19 +3,20 @@ import API from '../utils/api.js';
 import { getRevisionRange } from '../utils/mediawiki_revisions_utils';
 
 // This action uses the Thunk middleware pattern: instead of returning a plain
-// action object, it returns a function that takes the store dispatch fucntion —
+// action object, it returns a function that takes the store dispatch function —
 // which Thunk automatically provides — and can then dispatch a series of plain
 // actions to be handled by the store.
 // This is how actions with side effects — such as API calls — are handled in
 // Redux.
-export function fetchArticleDetails(articleId, courseId) {
-  return function (dispatch) {
+export function fetchArticleDetails(articleId, courseId, articleTitle, article_mw_page_id) {
+  return async function (dispatch) {
+    const crossCheckedArticleTitle = await dispatch(crossCheckArticleTitle(articleId, articleTitle, article_mw_page_id));
```

Comment:
Hmm... this will add an extra network call to every time we use fetchArticleDetails, even though in most cases the page has not moved.

Is it possible to implement this strategy only for cases where the query for article data doesn't work?

Reply:
> Is it possible to implement this strategy only for cases where the query for article data doesn't work?

Yes, I made the changes to  call `crossCheckedArticle` only if the last and first revisions are undefined, indicating that the article title has been moved.